### PR TITLE
edit profile pic feature added

### PR DESCRIPTION
**Pull request by SasiVardhan247**
https://github.com/narayan954/dummygram/assets/83742554/b58c4064-ed57-4830-acea-788245eaa8c6

This resolves issue #364 .

In video the upload folder is not recording here is ss:
Dummygram

### CODE REVIEW

File: src/pages/Profile.jsx [hunk 1]
```diff
@@ -9,16 +9,64 @@ import {
...
   };
 
+  const handleChange = (e) => {
+    if (e.target.files[0]) {
+      setAvatar(URL.createObjectURL(e.target.files[0]));
+      setImage(e.target.files[0]);
+      setVisibility(true);
+    }
+  };
+  console.log(name)
```
Comment (narayan954) [inline src/pages/Profile.jsx:36]: console logs are not required in prod

File: src/pages/Profile.jsx [hunk 2]
```diff
@@ -9,16 +9,64 @@ import {
...
+import { useState } from "react";
+import {auth,storage,} from "../lib/firebase";
+import { useSnackbar } from "notistack";
 
 function Profile() {
   const { name, email, avatar } = useLocation().state;
   const isNonMobile = useMediaQuery("(min-width: 768px)");
+  const { enqueueSnackbar } = useSnackbar();
+  const [image, setImage] = useState("");
+  const [img, setAvatar] = useState(avatar);
```
Comment (narayan954) [inline src/pages/Profile.jsx:21]: usestate should be declare as avatar,setAvatar or img,setImg

File: src/pages/Profile.jsx [hunk 3]
```diff
@@ -9,16 +9,64 @@ import {
...
+  const [visible,setVisibility] = useState(false);
```
Comment (narayan954) [inline src/pages/Profile.jsx:22]: usestate should be declare as visible,setVisible or visibility,setVisibility

File: src/pages/Profile.jsx [hunk 4]
```diff
@@ -9,16 +9,64 @@ import {
...
+      setVisibility(true);
+    }
+  };
+  console.log(name)
+  console.log(auth.currentUser.displayName)
+  const handleSave = async() => {
+    const uploadTask =  storage.ref(`images/${image?.name}`).put(image);
+   await uploadTask.on(
+      "state_changed",
+      () => {
```
Comment (narayan954) [inline src/pages/Profile.jsx:42]: do we need this function now?

File: src/pages/Profile.jsx (hunk 2 above)
Comment (narayan954) [reply src/pages/Profile.jsx:21]: or profilepic or anything meaningful

File: src/pages/Profile.jsx [hunk 5]
```diff
@@ -69,6 +131,13 @@ function Profile() {
           >
             Back
           </Button>
+          {visible && (<Button
```
Comment (narayan954) [inline src/pages/Profile.jsx:134]: move the save button below or above edit profile pic(perhaps above is better ux)

**Review by narayan954 — CHANGES_REQUESTED** [review]
some minor changes requested

File: src/pages/Profile.jsx (hunk 1 above)
Comment (SasiVardhan247) [reply src/pages/Profile.jsx:36]: changed

File: src/pages/Profile.jsx (hunk 2 above)
Comment (SasiVardhan247) [reply src/pages/Profile.jsx:21]: changed

File: src/pages/Profile.jsx (hunk 3 above)
Comment (SasiVardhan247) [reply src/pages/Profile.jsx:22]: changed

File: src/pages/Profile.jsx (hunk 4 above)
Comment (SasiVardhan247) [reply src/pages/Profile.jsx:42]: i took reference from signup page so i left it the same

File: src/pages/Profile.jsx (hunk 5 above)
Comment (SasiVardhan247) [reply src/pages/Profile.jsx:134]: done

File: src/pages/Profile.jsx (hunk 4 above)
Comment (SasiVardhan247) [reply src/pages/Profile.jsx:42]: is it the anonymous function in line 42 or the uploadTask.on function , with the uploadTask function i'm updating in the database.

Comment (narayan954) [reply src/pages/Profile.jsx:42]: anonymous function, which is basically empty

Comment (SasiVardhan247) [reply src/pages/Profile.jsx:42]: (snapshot) => {
        const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
        setProgress(progress);
      }
I thought of adding it for progress function

Comment (narayan954) [reply src/pages/Profile.jsx:42]: that'd require more implementation. Let's leave it for now

**Comment by narayan954** [comment]
When you're done, push the changes and request a review :)

**Comment by SasiVardhan247** [comment]
while pushing i'm getting conflicts so i'm closing this pr @narayan954

**Comment by narayan954** [comment]
fine

**Pull request closed without merging**